### DISCUSSION TITLE
Counting usage of DAX Measures in Excel?

### DISCUSSION THREAD

**Discussion by katiegainous** [DaxStudio/DaxStudio#697] [Q&A]
Hi there! 

Is it possible to count how many times a measure is used within a single Excel workbook/file? 

I have a workbook that is getting pretty large and I'd like to delete any obsolete / unused measures. 

Thanks!

**Comment by dgosbell** [comment]
This is a good idea and it has come up before, gut there is not currently an automated way of doing this with DAX Studio. You could turn on the All Queries trace then refresh all of your pivot tables and charts then export the trace to a json file and search through that... Although measures don't impact on the size of your model - it is unused tables and columns that impact on the size. 

And there are 2 tricky parts with this doing this sort of thing.

First it's hard to make sure that you have captured the full set of all possible queries. This is actually not too bad with a single PowerPivot file, but if that file is shared on Sharepoint or if we look at a Power BI file or SSAS model it becomes a lot harder as there could be many different reports and files referencing the one model. Because people can save reports on their local PCs there is no way to 100% guarantee that all usages have been captured. But there probably are use cases where you can be fairly confident that you've captured a good sample of queries.

Secondly we would also need to trace all the indirect references through measure expressions and things like sort by columns to make sure a particular column/measure was not referenced. It would be possible to build something that could do this, but we don't have anything like this at the moment and it's a pretty complex problem. This is a problem I'd like to solve, I just don't have any time line on when its likely to happen.
